client/components/Calendar.spec.js

```diff
@@ -22,16 +23,37 @@ describe('<Calendar/>', () => {
       })
 
         listing = JSON.parse(window.location.pathname.slice(1, -1));
-        state = { listingId: listing }
+        state = { listingId: listing, monthName: "Jun 2020", availableDates: [{"date":"2020-06-20","fee":39},{"date":"2020-06-20","fee":47},{"date":"2020-06-21","fee":54},{"date":"2020-06-21","fee":39},{"date":"2020-06-22","fee":39},{"date":"2020-06-23","fee":39},{"date":"2020-06-24","fee":39},{"date":"2020-06-24","fee":39}], allDates: [{"date":"2020-06-20","fee":39},{"date":"2020-06-20","fee":47},{"date":"2020-06-21","fee":54},{"date":"2020-06-21","fee":39},{"date":"2020-06-22","fee":39},{"date":"2020-06-23","fee":39}]};
     })
 
-    it('should load with the proper listingId', async (done) => {
+    it(`should render with default props`, async (done) => {
+      const wrapper = shallow(<Calendar listingData={state}/>);
+      expect(wrapper).toMatchSnapshot();
+      done()
+    });
 
-        const wrapper = mount(<Calendar listingData={state}/>)
+    it('should render at least 6 Pages of dates', async (done) => {
+      const wrapper = shallow(<Calendar listingData={state}/>)
+      expect(wrapper.find(Page).length).toBeGreaterThanOrEqual(6);
+      done()
+    });
 
-        
+    it('should load with a Clear Dates button', async (done) => {
+
+        const wrapper = shallow(<Calendar listingData={state}/>)
         expect(wrapper.exists()).toBeTruthy();
-        expect(wrapper.props().listingData.listingId).toBe(54);
+        expect(wrapper.text()).toContain('Clear Dates');
         done();
+    });
+
+    it('should call clearDates when ClearDates is clicked', async (done) => {
```

Comment:
Why is this async?

Reply:
the tests?  why yes they are.  very similar to our databases project.